arm/Microsoft.Automation/automationAccounts/jobSchedules/deploy.bicep

```diff
@@ -0,0 +1,53 @@
+@description('Optional. Name of the Automation Account job schedule. Must be a GUID. If not provided, a GUID is generated leveraging runbookName and scheduleName.')
+param name string = ''
+
+@description('Required. Name of the parent Automation Account.')
+param parent string
+
+@description('Required. The runbook property associated with the entity.')
+param runbookName string
+
+@description('Required. The schedule property associated with the entity.')
+param scheduleName string
+
+@description('Optional. List of job properties.')
+param parameters object = {}
+
+@description('Optional. The hybrid worker group that the scheduled job should run on.')
+param runOn string = ''
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered.')
+param cuaId string = ''
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource jobSchedule_automationAccount 'Microsoft.Automation/automationAccounts@2020-01-13-preview' existing = {
+  name: parent
+}
+
+resource jobSchedule 'Microsoft.Automation/automationAccounts/jobSchedules@2020-01-13-preview' = {
+  name: empty(name) ? guid(runbookName, scheduleName, jobSchedule_automationAccount.id) : name
+  parent: jobSchedule_automationAccount
```

Comment:
That's a cool idea. We should probably align all modules to it, instead of using string concat

Reply:
Would probably suggest a generic name though. E.g. `resource_parent` or just `parent` instead of `jobSchedule_automationAccount`

Reply:
The name comes from the existing resource referenced above. I used the convention we have for referenced modules and resources i.e. `<mainResource>_<referencedResource>`. If that confuses we may use the name of the parent resource e.g. `automationAccount`. Not sure about a generic name.

Reply:
In this particular case I was more thinking about how we handled e.g. rbac as it was the same for every module. We essentially use `roleAssignment` everywhere as a resource name, because the concept is the same across all modules. Here it would be the same. However, your point is just as valid. Let's discuss offline and just decide on something.

Reference:
```bicep
resource container 'Microsoft.Storage/storageAccounts/blobServices/containers@2019-06-01' existing = {
  name: resourceName
}

resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignmentObj.principalIds: {
  name: guid(container.name, principalId, roleAssignmentObj.roleDefinitionIdOrName)
  properties: {
    roleDefinitionId: contains(builtInRoleNames, roleAssignmentObj.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignmentObj.roleDefinitionIdOrName] : roleAssignmentObj.roleDefinitionIdOrName
    principalId: principalId
  }
  scope: container
}]
```

Reply:
Using `automationAccount` would be consistent with the rbac approach. If we used `parent` here we should use `parent` instead of `container` also in the rbac case. Anyway agree let's discuss offline

Reply:
Done